src/uwtools/set_config.py

```diff
@@ -39,50 +39,73 @@ def parse_args(argv):
         help='Path to a config base file.',
         required=True,
         type=path_if_file_exists,
-        )
+    )
+
+    parser.add_argument(
+        '--in_file_type',
+        help='Optional input file type',
+    )
 
     parser.add_argument(
         '-o', '--outfile',
         help='Full path to output file',
-        )
+    )
+
+    parser.add_argument(
+        '--out_file_type',
+        help='Optional output file type override',
+    )
 
     parser.add_argument(
         '-c', '--config_file',
         help='Optional path to configuration file',
         type=path_if_file_exists,
     )
 
+    parser.add_argument(
+        '--config_file_type',
+        help='Optional config file type override',
+    )
+
     parser.add_argument(
         '-d', '--dry_run',
         action='store_true',
         help='If provided, print rendered config file to stdout only',
     )
+
+    parser.add_argument(
+        '--values_needed',
+        action='store_true',
+        help='If provided, print a list of required configuration settings to stdout',
+    )
     return parser.parse_args(argv)
 
 
 def create_config_obj(argv):
     '''Main section for processing config file'''
 
     user_args = parse_args(argv)
-    file_type = get_file_type(user_args.input_base_file)
+    infile_type = user_args.in_file_type or get_file_type(user_args.input_base_file)
 
-    if file_type in [".yaml", ".yml"]:
+    if infile_type in [".yaml", ".yml"]:
         config_obj = config.YAMLConfig(user_args.input_base_file)
 
-    elif file_type in [".bash", ".sh", ".ini", ".IN"]:
+    elif infile_type in [".bash", ".sh", ".ini", ".IN"]:
         config_obj = config.INIConfig(user_args.input_base_file)
 
-    elif file_type == ".nml":
+    elif infile_type == ".nml":
         config_obj = config.F90Config(user_args.input_base_file)
 
+    elif infile_type == "field_table":
+        config_obj = config.FieldTableConfig()
+
     else:
         print("Set config failure: bad file type")
 
-
     if user_args.config_file:
-        config_file_type = get_file_type(user_args.config_file)
+        config_file_type = user_args.config_file_type or get_file_type(user_args.config_file)
 
-        if config_file_type in [".yaml", ".yml"]:
+        if config_file_type in [".yaml", ".yml", "YML", "field_table"]:
```

Comment:
We won't be able to parse a field_table.

Reply:
Ah, that's a remnant of the original pseudocode, I had meant to remove it - thanks for catching me!